gcputil/credentials.go

```diff
@@ -172,24 +205,44 @@ func ServiceAccountPublicKey(serviceAccount string, keyId string) (interface{},
 	return PublicKey(kStr)
 }
 
-// OAuth2RSAPublicKey returns the PEM key file string for Google Oauth2 public cert for the given 'kid' id.
-func OAuth2RSAPublicKey(ctx context.Context, keyId string) (interface{}, error) {
-	certUrl := googleOauthProviderX509CertUrl
-	res, err := cleanhttp.DefaultClient().Get(certUrl)
+// OAuth2RSAPublicKey returns the public key with the given key ID from Google's
+// public set of OAuth 2.0 keys. If the key does not exist, an error is returned.
+func OAuth2RSAPublicKey(ctx context.Context, keyID string) (interface{}, error) {
+	return OAuth2RSAPublicKeyWithEndpoint(ctx, keyID, "")
+}
+
+// OAuth2RSAPublicKeyWithEndpoint returns the public key with the given key ID from
+// Google's public set of OAuth 2.0 keys. If endpoint is provided, it will be used as
+// the service endpoint for the request. If endpoint is not provided, a default of
+// "https://www.googleapis.com" will be used. If the key does not exist, an error is
+// returned.
+func OAuth2RSAPublicKeyWithEndpoint(ctx context.Context, keyID, endpoint string) (interface{}, error) {
+	if endpoint == "" {
+		endpoint = defaultGoogleAPIsEndpoint
+	}
+
+	certUrl := strings.TrimSuffix(endpoint, "/") + googleOAuthProviderX509CertURLPath
+	req, err := http.NewRequest(http.MethodGet, certUrl, nil)
 	if err != nil {
 		return nil, err
 	}
-	if err := googleapi.CheckResponse(res); err != nil {
+	resp, err := cleanhttp.DefaultClient().Do(req.WithContext(ctx))
```

Comment:
Nit: Instead of calling `req.WithContext(ctx)` you can call `http.NewRequestWithContext` provide the context when the object is created.

Reply:
Ah, that's right. Done in 88a934c 👍